src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -151,28 +236,27 @@ AmendmentTableImpl::get (std::string const& name)
     return uint256 ();
 }
 
-AmendmentState*
-AmendmentTableImpl::addKnown (const char* amendmentID, const char* friendlyName,
-    bool veto)
+bool
+AmendmentTableImpl::addKnown (uint256 const& hash,
+                              std::string const& friendlyName,
+                              bool veto)
 {
-    uint256 hash;
-    hash.SetHex (amendmentID);
-
     if (hash.isZero ())
     {
         assert (false);
-        return nullptr;
+        return false;
     }
 
-    AmendmentState* f = getCreate (hash, true);
+    ScopedLockType sl (mLock);
+    AmendmentState* s = getCreate (hash, true);
 
-    if (friendlyName != nullptr)
-        f->setFriendlyName (friendlyName);
+    if (!friendlyName.empty ())
+        s->setFriendlyName (friendlyName);
```

Comment:
Note that you're using `s` without checking that it's non-null.  Or if you know it's non-null, then you should use a reference instead.

(You might have the convention that mutable objects are always referred to by pointers, which was my pre-ripple custom, but we make free use of references here.  I confess that it's quite useful...)